tests/test_core.py

```diff
@@ -1639,6 +1639,20 @@ class Polymorphic {virtual void member(){}};
 }
 ''', 'exception caught: std::bad_typeid')
 
+  @with_both_eh_sjlj
+  def test_terminate_abort(self):
```

Comment:
How about renaming this "test_abort_no_dtors" and rewriting it to call C `abort`, with a C++ object on the stack? 

The reason I ask is that this change seem more about the behaviour of `abort` than the behaviour of the higher level `std::terminate`

Reply:
Done